docs/reference/action_list.rst

```diff
@@ -226,6 +226,115 @@ to more field types, see `SonataDoctrineORMAdminBundle Documentation`_.
     values so use the ``inverse`` option if you really cannot find a good enough
     antonym for the name you have.
 
+Symfony Data Transformers
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If the model field has a limited list of values (enumeration), it is convenient to use a value object to control
+the available values. For example, consider the value object of moderation status with the following values:
+``awaiting``, ``approved``, ``rejected``::
+
+    final class ModerationStatus
+    {
+        public const AWAITING = 'awaiting';
+        public const APPROVED = 'approved';
+        public const REJECTED = 'rejected';
+
+        private static $instances = [];
+
+        private string $value;
+
+        private function __construct(string $value)
+        {
+            if (!array_key_exists($value, self::choices())) {
+                throw new \DomainException(sprintf('The value "%s" is not a valid moderation status.', $value));
+            }
+
+            $this->value = $value;
+        }
+
+        public static function byValue(string $value): ModerationStatus
+        {
+            // limitation of count object instances
+            if (!isset(self::$instances[$value])) {
+                self::$instances[$value] = new static($value);
+            }
+
+            return self::$instances[$value];
+        }
+
+        public function getValue(): string
+        {
+            return $this->value;
+        }
+
+        public static function choices(): array
+        {
+            return [
+                self::AWAITING => 'moderation_status.awaiting',
+                self::APPROVED => 'moderation_status.approved',
+                self::REJECTED => 'moderation_status.rejected',
+            ];
+        }
+
+        public function __toString(): string
+        {
+            return self::choices()[$this->value];
+        }
+    }
+
+To use this Value Object in the _`Symfony Form`: https://symfony.com/doc/current/forms.html component, we need a
+_`Data Transformer`: https://symfony.com/doc/current/form/data_transformers.html ::
+
+    use Symfony\Component\Form\DataTransformerInterface;
+    use Symfony\Component\Form\Exception\TransformationFailedException;
+
+    final class ModerationStatusDataTransformer implements DataTransformerInterface
+    {
+        public function transform($value): ?string
+        {
+            $status = $this->reverseTransform($value);
+
+            return $status instanceof ModerationStatus ? $status->value() : null;
+        }
+
+        public function reverseTransform($value): ?ModerationStatus
+        {
+            if ($value === null || $value === '') {
```

Comment:
We're currently using `@Symfony` as part of our CS rules, which implies the `yoda_style` rule:
```suggestion
            if (null === $value || '' === $value) {
```

Reply:
@greg0ire recently said that the Yoda conditions is not very good and maybe you should give it up. https://github.com/sonata-project/SonataAdminBundle/pull/6062#discussion_r416909700
Do you still want to use Yoda conditions?

Reply:
At first check 
```php
if ($value instanceof ModerationStatus) {
    return $value;
}
```
Then value must be string.
```suggestion
            if (0 === strlen($value) {
```
You can also move it to try{} to avoid warning about wrong type.

Reply:
> @greg0ire recently said that the Yoda conditions is not very good and maybe you should give it up. [#6062 (comment)](https://github.com/sonata-project/SonataAdminBundle/pull/6062#discussion_r416909700)
> Do you still want to use Yoda conditions?

It's still the coding standard ATM. We won't drop it without some discussion and updating all the base code. So for now the doc should be similar to the base code IMHO.

Reply:
Ok. I change it